crates/sui-graphql-rpc/src/types/protocol_config.rs

```diff
@@ -58,68 +60,82 @@ impl ProtocolConfigs {
+            .clone()
             .into_iter()
-            .map(|(key, value)| ProtocolConfigAttr {
-                key,
-                value: value.map(|v| v.to_string()),
-            })
+            .map(|(key, value)| ProtocolConfigAttr { key, value })
             .collect()
     }
 
     /// Query for the value of the configuration with name `key`.
     async fn config(&self, key: String) -> Option<ProtocolConfigAttr> {
-        self.native
-            .attr_map()
-            .get(&key)
-            .map(|value| ProtocolConfigAttr {
-                key,
-                value: value.as_ref().map(|v| v.to_string()),
-            })
+        self.configs.get(&key).map(|value| ProtocolConfigAttr {
+            key,
+            value: value.as_ref().map(|v| v.to_string()),
+        })
     }
 
     /// Query for the state of the feature flag with name `key`.
     async fn feature_flag(&self, key: String) -> Option<ProtocolConfigFeatureFlag> {
-        self.native
-            .feature_map()
+        self.feature_flags
             .get(&key)
             .map(|value| ProtocolConfigFeatureFlag { key, value: *value })
     }
 }
 
 impl ProtocolConfigs {
     pub(crate) async fn query(db: &Db, protocol_version: Option<u64>) -> Result<Self, Error> {
-        use chain_identifier::dsl as c;
         use epochs::dsl as e;
+        use feature_flags::dsl as f;
+        use protocol_configs::dsl as p;
 
-        let (latest_version, digest_bytes): (i64, Option<Vec<u8>>) = db
+        let latest_version: i64 = db
             .execute(move |conn| {
                 conn.first(move || {
                     e::epochs
-                        .select((
-                            e::protocol_version,
-                            c::chain_identifier
-                                .select(c::checkpoint_digest)
-                                .single_value(),
-                        ))
+                        .select(e::protocol_version)
                         .order_by(e::epoch.desc())
                 })
             })
             .await
-            .map_err(|e| Error::Internal(format!("Failed to fetch system details: {e}")))?;
-
-        let native = NativeProtocolConfig::get_for_version_if_supported(
-            protocol_version.unwrap_or(latest_version as u64).into(),
-            ChainIdentifier::from_bytes(digest_bytes.unwrap_or_default())?.chain(),
-        )
-        .ok_or_else(|| {
-            Error::ProtocolVersionUnsupported(
-                ProtocolVersion::MIN.as_u64(),
-                ProtocolVersion::MAX.as_u64(),
-            )
-        })?;
-
-        Ok(ProtocolConfigs { native })
+            .map_err(|e| {
+                Error::Internal(format!(
+                    "Failed to fetch latest protocol version in db: {e}"
+                ))
+            })?;
+
+        let version = protocol_version.unwrap_or(latest_version as u64);
+
+        let configs: BTreeMap<String, Option<String>> = db
```

Comment:
Is there a neat way to get all this data in a single DB query? We should at least re-use the connection.

Reply:
We could implement a DataLoader for this, but that's not a blocker for this PR.